mule-extensions-api/src/main/java/org/mule/runtime/extension/api/introspection/declaration/type/annotation/DisplayTypeAnnotation.java

```diff
@@ -21,29 +25,52 @@
 
   private final DisplayModel displayModel;
 
+  /**
+   * Creates a new instance
+   *
+   * @param displayModel a {@link DisplayModel}
+   */
   public DisplayTypeAnnotation(DisplayModel displayModel) {
     this.displayModel = displayModel;
   }
 
+  /**
+   * {@inheritDoc}
+   */
   @Override
   public String getName() {
     return NAME;
   }
 
+  /**
+   *
+   * @return the name which should be use to render this model.
+   */
   public String getDisplayName() {
     return displayModel.getDisplayName();
   }
 
+  /**
+   *
+   * @return a brief overview about this model.
+   */
   public String getSummary() {
     return displayModel.getSummary();
   }
 
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof DisplayTypeAnnotation))
-      return false;
-    DisplayTypeAnnotation displayTypeAnnotation = ((DisplayTypeAnnotation) obj);
-    return displayTypeAnnotation.getDisplayName().equals(this.getDisplayName())
-        && displayTypeAnnotation.getSummary().equals(this.getSummary());
+    if (obj instanceof DisplayTypeAnnotation) {
+      DisplayTypeAnnotation displayTypeAnnotation = ((DisplayTypeAnnotation) obj);
+      return displayTypeAnnotation.getDisplayName().equals(this.getDisplayName())
+          && displayTypeAnnotation.getSummary().equals(this.getSummary());
+    }
+
+    return false;
+  }
+
+  @Override
+  public int hashCode() {
+    return displayModel.hashCode();
```

Comment:
this is wrong! hashCode should operate over the same criterias than the equals. DisplayModel doesn't redefine hashCode so this will throw a different value on each instance.

That been said, it's not a bad idea to delegate into the displayModel since this type annotation is nothing but a wrapper. But you should move both, the equals and hashCode implementation to the DisplayModel and implement both of them correctly